### PR TITLE
feat: user and mentor dashboard tour feature

### LINKED ISSUE #799: 🚀 [Feature Request] - Dashboard Tour
### Is your feature request related to a problem ? Please describe. ✍️

1. Enhancing the UI of adding session page form
2. Listed session UI needs to be fixed
3. button position

and mobile version opitmization

### Describe briefly how you would implement this enhancement.

```bash
Login to dashboaord
```

### Additional context or screenshots

Additional context

### Are you contributing for a program?

Other

### Code of Conduct

- [X] I have read and agree to the Contributing Guidelines of this project.

### PR DESCRIPTION
**Pull request by rishuraj1**
## Related Issue

Closes: #799

## Description of Changes
- added user and mentor dashboard tour features 
- used reacttour library   https://www.npmjs.com/package/@reactour/tour

## Checklist:

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [x] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

## Screenshots

https://github.com/anmode/grabtern-frontend/assets/47839626/cca5134b-fd52-43c9-bb7e-235aa7146f49

https://github.com/anmode/grabtern-frontend/assets/47839626/bb4181a2-d778-4f99-ba51-79f0ace79728

Please provide any necessary screenshots to illustrate the changes made.

### CODE REVIEW

File: pages/dashboard/mentor.page.js [hunk 1]
```diff
@@ -50,14 +53,128 @@ function MentorDashboard() {
...
+    const mentorTour = localStorage.getItem("mentorTour");
+    if (mentorTour) return;
+    setIsOpen(true);
+
+    setSteps([
+      {
+        selector: "#mentorHome",
+        content: (
+          <div className="tw-flex tw-flex-col tw-items-center tw-justify-center">
+            <div className="tw-flex tw-justify-center tw-items-center">
```
Comment (anmode) [inline pages/dashboard/mentor.page.js]: can you please move this to dashboard components as tour component!? and it should be shows only once to user/mentor
you can use local storage for this!

File: pages/dashboard/user.page.js [hunk 2]
```diff
@@ -57,9 +60,81 @@ function userDashboard() {
...
 
+  //guided tour
+
+  const { setSteps, setCurrentStep, setIsOpen } = useTour();
+
+  useEffect(() => {
+    setCurrentStep(0);
+    const tour = localStorage.getItem("tour");
+    if (tour) return;
+    setIsOpen(true);
```
Comment (anmode) [inline pages/dashboard/user.page.js]: same this

**Review by anmode — CHANGES_REQUESTED** [review]
All looks good!

**Comment by anmode** [comment]
can we give option to skip tour?? but publish profile one is neccessary!

File: pages/dashboard/mentor.page.js (hunk 1 above)
Comment (rishuraj1) [reply pages/dashboard/mentor.page.js]: alright

**Comment by rishuraj1** [comment]
> can we give option to skip tour?? but publish profile one is neccessary!

the skip tour option is already given as cross button on the top right of the dialog box, and I don't think that we will be able to show only one step...but I will see again.

**Comment by anmode** [comment]
> > can we give option to skip tour?? but publish profile one is neccessary!
> 
> the skip tour option is already given as cross button on the top right of the dialog box, and I don't think that we will be able to show only one step...but I will see again.

oh ok my bad!

**Comment by rishuraj1** [comment]
@anmode I might need some help as I am facing difficulties storing the state and displaying the tour according to it.

So, when a user or mentor logs in to the dashboard the `tour` is displayed and the `userTour` (in case of user)  and `mentorTour` (in case of a mentor) variables are saved with value `true`. So when we reload the page the tour doesn't show up again.... But when user/mentor logs out and re logs in, again the tour is displayed.

**Comment by anmode** [comment]
> @anmode I might need some help as I am facing difficulties storing the state and displaying the tour according to it.
> 
> So, when a user or mentor logs in to the dashboard the `tour` is displayed and the `userTour` (in case of user) and `mentorTour` (in case of a mentor) variables are saved with value `true`. So when we reload the page the tour doesn't show up again.... But when user/mentor logs out and re logs in, again the tour is displayed.

i got it! during logout the localStorage got erased so we have to make it not to erase...

**Comment by anmode** [comment]
> @anmode I might need some help as I am facing difficulties storing the state and displaying the tour according to it.
> 
> So, when a user or mentor logs in to the dashboard the `tour` is displayed and the `userTour` (in case of user) and `mentorTour` (in case of a mentor) variables are saved with value `true`. So when we reload the page the tour doesn't show up again.... But when user/mentor logs out and re logs in, again the tour is displayed.

i got it! during logout the localStorage got erased so we have to make it not to erase...

**Comment by rishuraj1** [comment]
@anmode Okay, so will you do that please, meanwhile I am working on to switch background color when it is in dark mode

**Comment by rishuraj1** [comment]
@anmode , I am sorry Anmol I don't get it so can you please fix that.

**Comment by anmode** [comment]
> @anmode , I am sorry Anmol I don't get it so can you please fix that.

haaa no wories i'll do that...I hp tour is mobile optimized. however you can create final PR for all dashboard fix you might encounter! it would be great if you can test by creating a mentor account and using your dashboard thanks!

**Pull request merged**